vignettes/tcplfit2-vignette.Rmd

```diff
@@ -362,3 +362,148 @@ The output table resulting from the previous code chunk is the same format as th
+```{r example 5 upper}
+# load example data
+spid <- unique(mc3$spid)[26]
+temp <- mc3[is.element(mc3$spid,spid),]
+
+# The data file has stored concentration in log10 form, so fix that
+conc <- 10**temp$logc # back-transforming concentrations on log10 scale
+resp <- temp$resp
+
+# pull out all of the chemical identifiers and the name of the assay
+dtxsid <- temp[1,"dtxsid"]
+casrn <- temp[1,"casrn"]
+name <- temp[1,"name"]
+assay <- temp[1,"assay"]
+
+# create the row object
+row <- list(conc = conc, resp = resp, bmed = 0, cutoff = cutoff, onesd = onesd,assay=assay,
+            dtxsid=dtxsid,casrn=casrn,name=name)
+
+# run the concentration-response modeling for a single sample
+res_up <- concRespCore(row,fitmodels = c("cnst", "hill", "gnls", "poly1", "poly2", 
+                                         "pow", "exp2", "exp3", "exp4", "exp5"), 
+                       conthits = T, aicc = F, bidirectional=F)
+
+concRespPlot(res_up,ymin=-10,ymax=100)
+```
+
+```{r example 5 upper-res, echo=FALSE}
+# max conc.
+res_up['Max Conc.'] <- max(conc)
+res_up['Name'] <- name
+res_up[1, c("Max Conc.", "bmd", "bmdl", "bmdu")] <- round(res_up[1, c("Max Conc.", "bmd", "bmdl", "bmdu")], 3)
+# function results
+DT::datatable(res_up[1, c('Name','Max Conc.', "bmd", "bmdl", "bmdu")],rownames = FALSE)
+```
+
+The BMD 299.927 is higher than the maximum concentration in the data, which is 80. One can set `bmd_up_bnd` to 1 to use the maximum concentration as the upper bound (i.e., BMD is not allowed to be larger than 100% of the maximum concentration in the data). And if BMD is bigger than this upper bound, it will be forced to be the maximum concentration. One can also input a fraction to `bmd_up_bnd` so BMD is not allowed to be larger than a percentage of the maximum concentration. For example, set it to 0.6 to have BMD no larger than 60% percentage of the maximum concentration. Doing this can avoid the maximum concentration to be the BMD and make sure BMD is in the dose range tested so the analysis can have some dose groups below the BMD and some dose groups above BMD.
+```{r example upper 2, eval=FALSE, echo=FALSE}
+# input a number larger than 1
+res_up2 <- concRespCore(row,fitmodels = c("cnst", "hill", "gnls", "poly1", "poly2", "pow", "exp2", "exp3", "exp4", "exp5"), conthits = T, aicc = F, bidirectional=F, bmd_up_bnd = 2)
+```
+
+```{r example 5 upper cont., echo=FALSE, eval=FALSE}
+res_up2['Max Conc.'] <- max(conc)
+DT::datatable(res_up2[1, c('Max Conc.',"bmd", "bmdl", "bmdu")],rownames = FALSE,options = list(scrollX = T))
+```
+
+```{r example upper-demo}
+res_up3 <- concRespCore(row,fitmodels = c("cnst", "hill", "gnls", "poly1", "poly2", 
+                                          "pow", "exp2", "exp3", "exp4", "exp5"), 
+                        conthits = T, aicc = F, bidirectional=F, bmd_up_bnd = 0.6)
```

Comment:
bmd_up_bnd must be >= 1 per discussion with Richard and other SME's this morning.  Please make sure to just update this part of the example accordingly.